pkg/handlers/scripts/install.sh

```diff
@@ -1,116 +1,357 @@
 #!/bin/bash
-TMP_DIR="/tmp/tmpinstalldir"
+
+# Installer script for {{ .Program }} version {{ .Release }}.
+#
+# This script will verify that the environment is suitable for installation before downloading
+# and installing {{ .Program }}.
+#
+# This script can be configured by either setting environment variables or using argument flags,
+# but the command line arguments take precedence over the environment variables.
+#
+# Environment variables:
+# ----------------------
+#   REPL_USE_SUDO          set this to any value to use sudo when writing to the installation directory.
+#
+#   REPL_INSTALL_PATH      alternative installation directory to use.
+#
+# Command line arguments:
+# -----------------------
+#   -i, --install  alternative installation directory to use.
+#   -s, --sudo     use sudo when writing to the installation directory.
+
+READ_TIMEOUT=15
+DEFAULT_DIR="/usr/local/bin"
+INSECURE="{{ .Insecure }}"
+PROG="{{ .Program }}"
+RELEASE="{{ .Release }}"
+TMP_DIR=$(mktemp -d -t replicated-XXXXXX)
+USER="{{ .User }}"
+
+# Check for and use any environment variables.
+if [[ ! -z "${REPL_INSTALL_PATH:+x}" ]]; then
+	OUT_DIR="${REPL_INSTALL_PATH/#~/${HOME}}"
+fi
+
+if [[ ! -z "${REPL_USE_SUDO:+x}" ]]; then
+	USE_SUDO=1
+fi
+
+function print_help {
+	echo "{{ .Program }} installer script."
+	echo ""
+	echo "USAGE:"
+	echo "   $(basename "$0") [OPTIONS]"
+	echo "   $(basename "$0") [OPTIONS] -i PATH"
+	echo "   $(basename "$0") [OPTIONS] -s"
+	echo ""
+	echo "ARGS:"
+	echo "   PATH      A directory to install into"
+	echo ""
+	echo "OPTIONS:"
+	echo "   -h --help              print this help message"
+	echo "   -i --install <PATH>    use PATH as the install directory"
+	echo "   -s --sudo              use sudo to install (interactive)"
```

Comment:
so I only just noticed this

`-s` is really not a shortcode we should use, as it's already a bash flag - it essentially means 'pass in these flags'